velox/docs/develop/types.rst

```diff
@@ -22,34 +22,41 @@ BIGINT                  int64_t                         8
 DATE                    struct Date                     8
 REAL                    float                           4
 DOUBLE                  double                          8
-SHORT_DECIMAL           struct UnscaledShortDecimal     8
-LONG_DECIMAL            struct UnscaledLongDecimal     16
+DECIMAL                 int64_t / int128_t              8 / 16
```

Comment:
Should we add HUGEINT type?

It might be clearer to add a paragraph at the start to explain that Velox has a fixed set of physical types and extensible set of logical types. For physical types, we could list these in a table and specify a C++ type and # of bytes used per value. For logical types, we would specify which physical type they are backed by. Similarly, for Presto type, we can clarify that these are all logical types and specify physical types they are backed by.

Reply:
I like the above classification!
However, Timestamp with Timezone is implemented by a RowType (`Row<int64_t, int16_t>`) which is not exactly a physical type. Probably we should introduce CustomTypes and then use that in the table.

Reply:
The presence of ROW makes it a RowType, which is a complex type?

Reply:
I'm thinking of scalar vs. complex types as a different dimension from physicial vs. logical. Physical type determines the in-memory representation, while logical types adds interpretation / semantics. ROW(BIGINT, SMALLINT) is a physical complex type, while TIMESTAMP WITH TIME ZONE is a logical type based by ROW(BIGINT, SMALLINT) physical type.

I do notice that it is intuitive to think of TIMESTAMP WITH TIME ZONE as scalar type and it is confusing that it is backed by complex type. I personally think it was a mistake to implement TIMESTAMP WITH TIME ZONE as ROW(BIGINT, SMALLINT). It would be nice to re-implement this type using BIGINT as a backing physical type.

Reply:
I went with ROW(BIGINT, SMALLINT) for now.